WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -15,45 +15,66 @@ public partial class LabelSelectionViewModel : ViewModelBase
 
 	[AutoNotify] private bool _enoughSelected;
 
-	private Pocket? _privatePocket;
+	private Pocket _privatePocket = Pocket.Empty;
 	private bool _includePrivatePocket;
+	private Pocket[] _allPockets = Array.Empty<Pocket>();
 
 	public LabelSelectionViewModel(Money targetAmount)
 	{
 		_targetAmount = targetAmount;
 	}
 
-	public Pocket[] AllPockets { get; set; } = Array.Empty<Pocket>();
+	public Pocket[] NonPrivatePockets { get; set; } = Array.Empty<Pocket>();
 
 	public IEnumerable<LabelViewModel> AllLabelsViewModel { get; set; } = Array.Empty<LabelViewModel>();
 
 	public IEnumerable<LabelViewModel> LabelsWhiteList => AllLabelsViewModel.Where(x => !x.IsBlackListed);
 
 	public IEnumerable<LabelViewModel> LabelsBlackList => AllLabelsViewModel.Where(x => x.IsBlackListed);
 
-	public Pocket[] GetSafeToUsePockets()
+	public Pocket[] AutoSelectPockets()
 	{
-		var pockets = AllPockets.ToList();
+		var knownPockets = NonPrivatePockets.Where(x => x.Labels != CoinPocketHelper.UnlabelledFundsText).ToArray();
+		var unknownPockets = NonPrivatePockets.Except(knownPockets).ToArray();
+		var privateAndUnknownPockets = _allPockets.Except(knownPockets).ToArray();
+		var privateAndKnownPockets = _allPockets.Except(unknownPockets).ToArray();
 
-		if (_privatePocket is { } privatePocket)
+		if (_privatePocket.Amount >= _targetAmount)
 		{
-			pockets.Add(privatePocket);
+			return new[] { _privatePocket };
 		}
 
-		var unlabelledPocket = pockets.FirstOrDefault(x => x.Labels == CoinPocketHelper.UnlabelledFundsText);
+		if (knownPockets.Sum(x => x.Amount) >= _targetAmount)
```

Comment:
I wonder about comparing to `_targetAmount` here. Should we compare against effective coin values here or not?

Reply:
What does effective coin value mean?

Reply:
`SmartCoin.Amount` gives you a value of some UTXO. However, if the UTXO is worth 1 BTC, you cannot really spend 1 BTC because you need to pay some fee.

See https://github.com/zkSNACKs/WalletWasabi/blob/3eac9ad39483dabe88b496ee1e26bbf82a67b2b9/WalletWasabi/Extensions/NBitcoinExtensions.cs#L477-L478 for relevant code.

Reply:
I don't know if it is of any concern here. I just want to make sure it was taken into acount if applicable.

Reply:
It hasn't been taken into account, but right now it isn't even possible. Needs a tiny refactoring to have the selected fee rate at this point in every case.

I will take a look, thanks to pointed this out.